### PR TITLE
DOC: mark mosaic as no longer provisional

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Mark mosaic as no longer provisional.

I also moved this from tutorials -> examples as this is much a how-to and "gallery" is what we have that matches the divio "how-to" best.

### CODE REVIEW

File: examples/subplots_axes_and_figures/mosaic.py [hunk 1]
```diff
@@ -1,14 +1,10 @@
 """
+.. redirect-from:: /gallery/subplots_axes_and_figures/mosaic
```
Comment (jklymak) [inline examples/subplots_axes_and_figures/mosaic.py:2]: I think you want: 

```suggestion
.. redirect-from:: /tutorials/provisional/mosaic
```

**Review by jklymak — APPROVED** [review]

File: examples/subplots_axes_and_figures/mosaic.py (hunk 1 above)
Comment (tacaswell) [reply examples/subplots_axes_and_figures/mosaic.py:2]: sigh, the dangers of grep + xargs + sed 🤦🏻

File: examples/subplots_axes_and_figures/mosaic.py [hunk 2]
```diff
@@ -1,14 +1,10 @@
 """
+.. redirect-from:: /tutorials/provisional/mosaic
+
 =======================================
 Complex and semantic figure composition
```
Comment (jklymak) [inline examples/subplots_axes_and_figures/mosaic.py:5]:
```suggestion
subplot_mosaic: Complex and semantic figure composition
```

**Review by jklymak — COMMENTED** [review]
If you search for "mosaic" on the rendered docs example page, you do not get this example showing up.  I think folks wanting a quick guide as to how subplot_mosaic works will want that in the title.

**Review by oscargus — APPROVED** [review]
Should there be some sort of release note for this? It may make sense for users to know that this has happened in another way than noting that the provisional note has been removed.

**Comment by oscargus** [comment]
I may argue that it is indeed a tutorial type of document here. One could possibly add an example or two, but considering the style and length of the documentation, it seems more like a tutorial than an example. But if others disagree, I will not block based on this.

File: examples/subplots_axes_and_figures/mosaic.py (hunk 2 above)
Comment (tacaswell) [reply examples/subplots_axes_and_figures/mosaic.py:5]: Good point!  I did it silghtly differently and adjusted the decoration so just push a new commit.

**Comment by rcomer** [comment]
>  it seems more like a tutorial than an example

Within the tutorials, I prefer the subject categories (Colors, Text) to the basic/intermediate/advanced categories.  The subject links help me navigate to the thing that I currently want to learn about, whereas I don't know the difficulty level of the thing until I've learned it.  I wonder if we could make a "layouts" tutorial category for mosaic, constrained layout, tight layout and "arranging multiple Axes in a Figure".  I appreciate that that is a little out of scope for this PR though.

**Comment by tacaswell** [comment]
If we use the terminology from https://documentation.divio.com/tutorials/ then this is not a tutorial because it is just N related examples all on one page.  To riff on their cooking analogy, this is a list of every way to cook and egg not instructions on how to make a nice omelette breakfast.  Under this rubric I think many of the things currently in tutorials are miss-categorized (https://matplotlib.org/stable/tutorials/introductory/lifecycle.html and maybe the image one may be the only ones that actually fit the divio definition).

Good point about the release notes, I'll add that.

**Comment by jklymak** [comment]
Certainly some of what we have in "tutorials" are not actual tutorials, but are long-form documentation of features and how the library works. I'd argue that 10 or 12 examples as part of a narrative of how a feature works is not really an "Example" either.  

Let's open a new issue to discuss how to organize the tutorials, since this has come up a few times.  Personally I wouldn't move the subplots_mosaic to Examples, but don't feel strongly about it.

**Pull request merged**